tests/hypervisor/test_default.py

```diff
@@ -0,0 +1,234 @@
+        """
+        :title: virt-who: default: test the mapping info
+        :id: 09e1754d-5f3a-49c5-aebc-91d4f4a8471e
+        """
+        guest_uuid = hypervisor_data['guest_uuid']
+        virt_type = {
+            'local': 'kvm',
+            'libvirt': 'kvm',
+            'rhevm': 'kvm',
+            'esx': 'vmware',
+            'hyperv': 'hyperv',
+            'xen': 'xen',
+            'kubevirt': 'kvm',
+            'ahv': 'nutanix_ahv'
+        }
+        # check virt.uuid fact by subscription-manager in guest")
+        cmd = "subscription-manager facts --list | grep virt.uuid"
+        _, output = ssh_guest.runcmd(cmd)
+        virt_uuid = output.split(':')[1].strip()
+        assert virt_uuid.lower() == guest_uuid.lower()
+
+        # check virt.host_type fact by subscription-manager in guest
+        _, virtwhat_output = ssh_guest.runcmd("virt-what")
+        _, facts_output = ssh_guest.runcmd(
+            "subscription-manager facts --list | grep virt.host_type")
+        assert (virt_type[HYPERVISOR] in virtwhat_output
+                and virt_type[HYPERVISOR] in facts_output)
+
+        # check virt.is_guest fact by subscription-manager in guest
+        cmd = "subscription-manager facts --list | grep virt.is_guest"
+        _, output = ssh_guest.runcmd(cmd)
+        virt_is_guest = output.split(':')[1].strip()
+        assert virt_is_guest == "True"
+
+    @pytest.mark.tier2
+    def test_delete_host_hypervisor(
+            self, virtwho, function_hypervisor, hypervisor_data, rhsm, satellite, register_data,
+            ssh_host, sm_host):
+        """
+        :title: virt-who: default: test the mapping info after deleting the host and hypervisor
+        :id: 19b84057-69a3-43bc-9b24-f39bc31ed3a8
+        :caseimportance: High
+        :tags: tier2
+        :customerscenario: false
+        :upstream: no
+        :steps:
+
+            1. Run the virt-who service to send the mapping info
+            2. Delete the virt-who host from webui
+            3. Run the virt-who service again to check the rhsm.log
+            3. Re-register the virt-who host and re-run the virt-who service
+            4. Delete the hypervisor from web UI,re-run the virt-who service
+        :expectedresults:
+
+            1. Virt-who works fine, no error messages in the rhsm.log
+            3. Virt-who works fine, can send the the mappping info
+            4. Virt-who Works fine, no error messages in the rhsm.log
+        """
+        host_name = hypervisor_data['hypervisor_hostname']
+        virtwho_hostname = hostname_get(ssh_host)
+
+        # run virt-who to send mappings
+        result = virtwho.run_service()
+        assert (result['error'] == 0
+                and result['send'] == 1
+                and result['thread'] == 1)
+
+        # delete virt-who host from webui
+        if REGISTER == 'rhsm':
+            rhsm.delete(virtwho_hostname)
```

Comment:
which should be changed to rhsm.host_delete

Reply:
Ok..., I just re-tested this after the rebase